### DISCUSSION TITLE
Issue with grpc conductor server using docker-compose

### DISCUSSION THREAD

**Discussion by chandruibm** [Netflix/conductor#2303] [Q&A]
I'm trying to use my sql DB for the persistence and used the docker/grpc/docker-compose. It starts the mysql, ES and ui, but the server fails with exit code 1. What do I need to configure inorder for this to work. The documentation doesn't seem to have any reference on how to setup using my sql or other persistence layer other than memory & redis. And it is becoming hard to understand what is expected to set up using mysql and how to resolve the problem

Steps to reproduce:
1. Clone the latest version of conductor -- 3.0.6
2. Go to docker/grpc
3. docker compsoe build
4. docker-compose up

It starts the following cotainer but not the server, and it logs this in console

Console Logs :
Creating volume "grpc_conductor_mysql" with default driver
Creating grpc_mysql_1         ... done
Creating grpc_elasticsearch_1 ... done
Creating grpc_conductor-server_1 ... done
Creating grpc_conductor-ui_1     ... done
Attaching to grpc_mysql_1, grpc_elasticsearch_1, grpc_conductor-server_1, grpc_conductor-ui_1
...........................
...........................
elasticsearch_1     | [2021-06-10T11:16:40,220][INFO ][o.e.c.m.MetaDataMappingService] [r-4p8ll] [.watches/jZhDBLWhSnWz4ll0T1_07A] update_mapping [watch]
grpc_conductor-server_1 exited with code 1
elasticsearch_1     | [2021-06-10T11:17:40,644][INFO ][o.e.c.m.MetaDataCreateIndexService] [r-4p8ll] [.triggered_watches] creating index, cause [auto(bulk api)], templates [triggered_watches], shards [1]/[1], mappings [triggered_watch]

Active containers shows only these and not the server, since it did exit:

CONTAINER ID   IMAGE                                                 COMMAND                  CREATED          STATUS                    PORTS                                                                                  NAMES
18a81da963fc   conductor:ui                                          "/bin/sh /app/startu…"   13 minutes ago   Up 13 minutes             0.0.0.0:5000->5000/tcp, :::5000->5000/tcp                                              grpc_conductor-ui_1
73cfae7c8716   docker.elastic.co/elasticsearch/elasticsearch:5.6.8   "/bin/bash bin/es-do…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:9200->9200/tcp, :::9200->9200/tcp, 0.0.0.0:9300->9300/tcp, :::9300->9300/tcp   grpc_elasticsearch_1
4612bcf8ec36   mysql:5.6                                             "docker-entrypoint.s…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:3306->3306/tcp, :::3306->3306/tcp                                              grpc_mysql_1

**Comment by kishorebanala** [comment]
Hey @michaelpaliy, Is this something you can help with?

@chandruibm Can you also share the start up logs for Conductor server please? It is most likely missing configuration for mysql module. Thanks!

**Comment by chandruibm** [comment]
Please find the server start up 
[server.log](https://github.com/Netflix/conductor/files/6655513/server.log)

logs

**Comment by michaelpaliy** [comment]
Hi @kishorebanala @chandruibm ,

There are some wrong configurations in the docker-compose file, for example:
1. The password in the config file doesn't match the DB password configured in the docker-compose file.
2. There is a missing link to elastic search from the conductor server.
3. the version of MySQL is no longer supported by Flyway Community Edition.

I fixed these issues and opened MR: https://github.com/Netflix/conductor/pull/2317

**Reply by chandruibm** [reply]
Thanks @michaelpaliy for addressing the issue. Bit of improvement in the logs.
Tried as per your changes and seems to getting problem with server

[35mmysql_1             |[0m 2021-06-17T08:40:01.129530Z 3 [Note] Access denied for user 'conductor'@'172.21.0.4' (using password: YES)
[35mmysql_1             |[0m 2021-06-17T08:40:01.338605Z 4 [Note] Got an error reading communication packets
[36mgrpc_conductor-server_1 exited with code 1
[0m[35mmysql_1             |[0m 2021-06-17T08:40:06.448650Z 5 [Note] Got an error reading communication packets


Attached the server.log for your reference.
Let me know if we need to provide permissions in mysql explicitly or will the configuration address them automatically.

[server.log](https://github.com/Netflix/conductor/files/6668706/server.log)

**Reply by michaelpaliy** [reply]
can you try to run the following command `docker volume rm grpc_conductor_mysql` and try again?
if it does not help try to rebuild the images.

**Reply by chandruibm** [reply]
@michaelpaliy  - Works like charm thank you after removing the volume and re-build.
Can we get a build of conductor-server-<version>-boot.jar with the above fix.
We extract the official build version from this location, hence when can this fix go in this repo+
https://repo1.maven.org/maven2/com/netflix/conductor/conductor-server/
